### DISCUSSION TITLE
next/link added aditional query string like 'path'

### DISCUSSION THREAD

**Discussion by sc-ruslanmatkovskyi** [vercel/next.js#68280] [Help]
### Summary

I noticed an unusual feature when using the Link component from next/link. When navigating via this link, a query string parameter path is added in the middleware, which interferes with request handling in the middleware. If you simply refresh the page, the path parameter is not present. How can this parameter be disabled? It prevents the middleware from being fully utilized.

Here's the code I'm using in my layout.tsx component:
`<Link href={{ pathname: '/about' }}>URL LINK</Link>`

I also tried adding an empty query value, but that didn't help either:
`<Link href={{ pathname: '/about', query: '' }}>URL LINK</Link>`

Additionally, if I add a path parameter in the query value, it gets overwritten:
`<Link href={{ pathname: '/about', query: { path: '123' } }}>URL LINK</Link>`

In the middleware, I get(req: NextRequest):
```
{
  href: 'http://localhost:3000/about?path=about',
  origin: 'http://localhost:3000',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'localhost:3000',
  hostname: 'localhost',
  port: '3000',
  pathname: '/about',
  search: '?path=about'
  ...
}

```
Is there a way to prevent the path parameter from being added when using the Link component? This issue is hindering the proper functioning of the middleware.

### Additional information

```js
{
  href: 'http://localhost:3000/about?path=about',
  origin: 'http://localhost:3000',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'localhost:3000',
  hostname: 'localhost',
  port: '3000',
  pathname: '/about',
  search: '?path=about',
  searchParams: <ref *1> URLSearchParams {
  [Symbol(query)]: [ 'path', 'about' ],
  [Symbol(context)]: URL {
  [Symbol(context)]: URLContext {
  href: 'http://localhost:3000/about?path=about',
  protocol_end: 5,
  username_end: 7,
  host_start: 7,
  host_end: 16,
  pathname_start: 21,
  search_start: 27,
  hash_start: 4294967295,
  port: 3000,
  scheme_type: 0
},
  [Symbol(query)]: [Circular *1]
}
},
  hash: ''
}
```

**Comment by ijjk** [comment]
> Is there a way to prevent the path parameter from being added when using the Link component? This issue is hindering the proper functioning of the middleware

Hi, yes if you want that query to only be used internally on the client and not added to the URL so that it's not there when refreshing then in pages router you can use `href`/`as` combination on `next/link` e.g. 

```jsx
<Link href={{ pathname: '/about', query: { path: '123' } }} as={{ pathname: '/about' }}></Link>
```

If you are in app router you can't assign hidden query params like this and it's either in the URL or not. 

Hope this answers the question, if not a minimal repro would shed better light on what you're trying to achieve.
